content/en/docs/tasks/security/authz-http/index.md

```diff
@@ -84,7 +84,7 @@ the services must have one of the listed `app` labels.
       - services: ["*"]
         methods: ["GET"]
         constraints:
-        - key: "destination.labels[app]"
+        - key: "source.labels[app]"
```

Comment:
Could you clarify why make this change? This doesn't match the description above: 
```
allows read access to any service in the default namespace that has the app label set to one of the values productpage, details, reviews, or ratings.
```

Reply:
Isn't that what the constraint is defining?  If I have the constraint that says a pod with the source label app = productpage, details, reviews, or ratings, allow access to the the downstream service, in this case `[*]` in the `default` namespace.  

Unless I'm understanding the path flow wrong `services: [*]` defines the service I want to protect and the constraint helps me define which service I want to allow access.

I tried to re-create this for one my scenarios of protecting Service A from Service B and using destination label didn't work until I switched it to source label.

Reply:
the constraint is actually used to further filter the services you want to protect. For example, you have `services: [*]` and constraints `destination.labels[app]: [dev]`, that means only the service with label `app: dev` is to be protected by the RBAC, another service that doesn't have label `app: dev` will not be protected by RBAC in this case.

Also note that the `source.labels` should only be used in the Properties: https://istio.io/docs/reference/config/authorization/constraints-and-properties/#supported-properties

Reply:
In the `ServiceRole`, isn't for granting access.  i.e. I have `service: ["serviceb"]`, but only a service with label `app:servicea` would be able to invoke it.

The definition of what services to be protected are `ClusterRbacConfig`, with `ON_WITH_INCLUSION`.   `services: ["serviceb.default.svc.cluster.local"]`

Unless I'm misusing the purpose.  For instance I used `destination.labels` for my use case, but it didn't work until I switched to `source.labels`

Reply:
`ClusterRbacConfig` is used to enable RBAC on a set of namespaces or services and ServiceRole is used to define the detailed RBAC policy on the namespace/service.

let's say you have svcA, svcB and svcC, and you have ClusterRbacConfig on ["SvcA", "SvcB"], this means if you don't define any ServiceRole/Binding for SvcA/SvcB, the request will be rejected. Here SvcC will always be accessible.

Now let's say you have ServiceRole with `services: [*]` with constrainst `destination.labels[app]: A`, this means it will only affect SvcA as the constraint is only matched with SvcA, the results is the request to SvcB will always be rejected because there is no policy selecting it.

As for constraint `source.labels`, it's not an supporter attribute, see https://istio.io/docs/reference/config/authorization/constraints-and-properties/#supported-properties

There might be something else wrong that makes you think it somehow worked in your case, would you mind to provide more context of your use case?